drivers/i2s/i2s_mcux_sai.c

```diff
@@ -431,72 +533,13 @@ static int i2s_mcux_config(const struct device *dev, enum i2s_dir dir,
 
 	memset(&config, 0, sizeof(config));
 
-	if (i2s_cfg->options & I2S_OPT_FRAME_CLK_SLAVE) {
-		if (i2s_cfg->options & I2S_OPT_BIT_CLK_SLAVE) {
-			config.masterSlave = kSAI_Slave;
-		} else {
-			config.masterSlave =
-				kSAI_Bclk_Master_FrameSync_Slave;
-		}
-	} else {
-		if (i2s_cfg->options & I2S_OPT_BIT_CLK_SLAVE) {
-			config.masterSlave =
-				kSAI_Bclk_Slave_FrameSync_Master;
-		} else {
-			config.masterSlave = kSAI_Master;
-		}
-	}
+	const bool is_mclk_master = i2s_cfg->options & I2S_OPT_BIT_CLK_MASTER;
```

Comment:
This line is broken because `I2S_OPT_BIT_CLK_MASTER` is 0, so this is always zero.
https://github.com/zephyrproject-rtos/zephyr/blob/e9b288b7137e5fb6ded18334aaa5a5f9c23ebb17/include/zephyr/drivers/i2s.h#L177

This needs to be a check for `i2s_cfg->options & I2S_OPT_BIT_CLK_SLAVE` or something similar.